### ISSUE TITLE
About validation set

### ISSUE THREAD

**Issue by ruizhaogit** [tflearn/tflearn#187]
Hi TFlearn team,

I have a line of code like this: model.fit(trainX, trainY, n_epoch=200, shuffle=None, validation_set=0.2)

How does TFlearn choose 20% from trainning set as validation, like randomly? And during the training this 20% validation set is fixed, right? 

One more question, normally, how much percent should be split as validation set? like 10 to 20%?

Than you!

Best,
Rui

**Comment by aymericdamien** [comment]
Yes, when you apply validation_split, your data will be shuffled, and a part will be used for validation.
There is no rules to choose your validation split, but usually people are using 10%.

**Comment by pankap** [comment]
Greetings to all TFLearn team and thank you for this great work. I would like to ask, whether your framework supports 10-fold cross validation when using 10% of the each class samples for validation.

**Comment by 0fork** [comment]
I'm not sure if I understand correctly how this library works regarding this so please bare with me :)

I'm training a CNN with little over 10 million images. This means that with 0.1 my validation set is around 100k images. When I get to training and hit a snapshot threshold it looks like that currently it just wants to test all 100k images on every single snapshot which of course is insanity because it takes a loooonnnggg time.

How can I tweak how many images get picked for testing in each snapshot validation?
